packages/react-aria-components/test/AriaAutocomplete.test-util.tsx

```diff
@@ -0,0 +1,561 @@
+
+          await user.keyboard('{ArrowDown}');
+          expect(input).toHaveAttribute('aria-activedescendant', options[0].id);
+          await user.keyboard('{ArrowDown}');
+          expect(input).toHaveAttribute('aria-activedescendant', options[1].id);
+          await user.keyboard('{ArrowDown}');
+          expect(input).toHaveAttribute('aria-activedescendant', options[2].id);
+          await user.keyboard('{ArrowUp}');
+          expect(input).toHaveAttribute('aria-activedescendant', options[1].id);
+
+          expect(document.activeElement).toBe(input);
+        });
+
+        it('should clear the focused key when using ArrowLeft and ArrowRight', async function () {
+          let {getByRole} = renderers.standard({});
+          let input = getByRole('searchbox');
+          let menu = getByRole(collectionNodeRole);
+          expect(input).not.toHaveAttribute('aria-activedescendant');
+
+          await user.tab();
+          expect(document.activeElement).toBe(input);
+
+          await user.keyboard('Foo');
+          act(() => jest.runAllTimers());
+          let options = within(menu).getAllByRole(collectionItemRole);
+          expect(input).toHaveAttribute('aria-activedescendant', options[0].id);
+          await user.keyboard('{ArrowRight}');
+          expect(input).not.toHaveAttribute('aria-activedescendant');
+          await user.keyboard('{ArrowDown}');
+          expect(input).toHaveAttribute('aria-activedescendant', options[0].id);
+          await user.keyboard('{ArrowLeft}');
+          expect(input).not.toHaveAttribute('aria-activedescendant');
+          expect(document.activeElement).toBe(input);
+        });
+
+        if (collectionType === 'menu') {
+          it('should trigger the wrapped element\'s onAction when hitting Enter', async function () {
+            let {getByRole} = renderers.standard({collectionProps: {onAction}});
+            let input = getByRole('searchbox');
+            let menu = getByRole(collectionNodeRole);
+            expect(input).not.toHaveAttribute('aria-activedescendant');
+
+            await user.tab();
+            expect(document.activeElement).toBe(input);
+
+            await user.keyboard('{ArrowDown}');
+            let options = within(menu).getAllByRole(collectionItemRole);
+            expect(input).toHaveAttribute('aria-activedescendant', options[0].id);
+            await user.keyboard('{Enter}');
+            expect(onAction).toHaveBeenCalledTimes(1);
+            expect(onAction).toHaveBeenLastCalledWith('1');
+
+            await user.keyboard('{ArrowDown}');
+            await user.keyboard('{Enter}');
+            expect(onAction).toHaveBeenCalledTimes(2);
+            expect(onAction).toHaveBeenLastCalledWith('2');
+          });
+
+          it('should not trigger the wrapped element\'s onAction when hitting Space', async function () {
+            let {getByRole} = renderers.standard({collectionProps: {onAction}});
```

Comment:
onAction is implementation specific and should be in a test for RAC, not in the aria tests. if you want, you can specify a renderer that is, "when item is clicked x happens", then assert for the x
that way, if someone makes an implementation where instead of `onAction` they have `onFlorbigook`, that will still pass

we should try very hard not to pass any props to the renderers because they more than likely bind the test to a specific API

Reply:
good call, I guess I conflated the fact that `onAction` is in our hook api with these general tests. I'll see what I can do to preserve these but make the onAction generic